module/x/gravity/keeper/grpc_query.go

```diff
@@ -135,6 +135,14 @@ func (k Keeper) BatchTxs(c context.Context, req *types.BatchTxsRequest) (*types.
 	return &types.BatchTxsResponse{Batches: batches, Pagination: pageRes}, nil
 }
 
+func (k Keeper) LastBatchTx(c context.Context, req *types.LastBatchTxRequest) (*types.LastBatchTxResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+	contract := common.HexToAddress(req.TokenContract)
+
+	batch := k.getLastOutgoingBatchByTokenType(ctx, contract)
```

Comment:
can this fail if the token contract address invalid?

Reply:
seems not, it will just crop the byte array to the correct length.

if token contract cannot be found, it will return nil